src/App.js

```diff
@@ -1,23 +1,25 @@
-import logo from './logo.svg';
+import Header from "./components/header";
+import Articles from "./components/articles";
+import {Routes, Route} from 'react-router-dom';
 import './App.css';
+import { useState, useEffect } from "react";
+import { fetchArticles } from './api';
 
 function App() {
+  const [articles, setArticles] = useState([]);
+
+  useEffect(() => {
+    fetchArticles().then((data) => {
+    setArticles(data);
+    })
+}, );
```

Comment:
At the moment there's an infinite loop here and it keeps fetching data continuously. Please include a dependency array in useEffect() as a second parameter.